functies.php

```diff
@@ -15,27 +15,58 @@ function getKabinet(): string
 
 function getBitcoinPrice(): string
 {
-    $url = 'https://api.coinbase.com/v2/exchange-rates?currency=BTC';
+    $url = 'https://api.kucoin.com/api/v1/market/stats?symbol=BTC-EUR';
     $jsonData = file_get_contents($url);
+
+    // Check if the data was fetched correctly
+    if ($jsonData === false) {
+        return 'Error fetching data from kucoin';
+    }
+
     $response = json_decode($jsonData);
 
-    return 'Bitcoin prijs: € '.number_format($response->data->rates->EUR, 2, ',', '.');
+    // Check if the JSON was decoded correctly
+    if ($response === null) {
+        return 'Error decoding JSON data';
+    }
+
+    // Ensure we are accessing the correct properties
+    if (! isset($response->data->last)) {
+        return 'Error: Last rate not found in response';
+    }
+
+    $bitcoinPrice = $response->data->last;
+
+    // Return the formatted price
+    return 'Bitcoin prijs: € '.number_format($bitcoinPrice, 2, ',', '.');
 }
 
 function getEthereumPrice(): string
 {
-    $url = 'https://api.coinbase.com/v2/exchange-rates?currency=ETH';
+    $url = 'https://api.kucoin.com/api/v1/market/stats?symbol=ETH-EUR';
     $jsonData = file_get_contents($url);
+
+    // Check if the data was fetched correctly
+    if ($jsonData === false) {
+        return 'Error fetching data from kucoin';
+    }
+
     $response = json_decode($jsonData);
 
-    if ($response && isset($response->data->rates->EUR)) {
```

Comment:
Bij bitcoin heb je deze in 2 if's gedaan en hier bij elkaar. Welke is beter?

Reply:
dit is een comment op de oude code

Reply:
Ja, ik kan hem ook op het nieuwe stuk doen. Maar 2 oplossingen voor hetzelfde probleem lijkt me wat raar.